torch/_inductor/codecache.py

```diff
@@ -761,25 +761,55 @@ def get_include_and_linking_paths(
         ipaths = cpp_extension.include_paths(cuda) + [sysconfig.get_path("include")]
         lpaths = []
         if sys.platform == "darwin":
-            # GNU OpenMP generally is not available on MacOS
-            # There is either Intel OpenMP(for x86) or LLVM OpenMP (for both x86 and arm64)
-            libs = ["omp"]
-            if os.getenv("OMP_PREFIX") is not None:
-                # Support OpenMP on MacOS
-                ipaths.append(os.path.join(os.getenv("OMP_PREFIX"), "include"))
-                lpaths.append(os.path.join(os.getenv("OMP_PREFIX"), "lib"))
-
-            if os.getenv("CONDA_PREFIX") is not None:
-                # On MacOS OpenMP is not available via the system install
-                # But on conda can be provided using https://anaconda.org/anaconda/llvm-openmp
-                conda_lib_path = os.path.join(os.getenv("CONDA_PREFIX"), "lib")
-                ipaths.append(os.path.join(os.getenv("CONDA_PREFIX"), "include"))
-                lpaths.append(conda_lib_path)
-                # Prefer Intel OpenMP on x86 machine
-                if os.uname().machine == "x86_64" and os.path.exists(
-                    os.path.join(conda_lib_path, "libiomp5.dylib")
-                ):
-                    libs = ["iomp5"]
+            cxx = cpp_compiler()
+            version_string = subprocess.check_output([cxx, "--version"]).decode('utf8')
+            # only Apple builtin compilers (Apple Clang++) require openmp
+            omp_available = 'Apple' not in version_string.splitlines()[0]
+
+            libs = [] if omp_available else ["omp"]
+
+            # check the `OMP_PREFIX` environment first
+            if not omp_available and os.getenv("OMP_PREFIX") is not None:
```

Comment:
I'd expect `OMP_PREFIX` to add flags even if `omp_available=True`, since the user might want to use a custom openmp (for example the intel one).

Reply:
Do you want this to hold for macos only, or for all os?
Currently the `OMP_PREFIX` is only intended to help macos users find the openmp.
What you propose seems to enable users to override the default openmp for all the operating systems.

Reply:
I have enabled the feature you want for macos in [9391468](https://github.com/pytorch/pytorch/pull/107111/commits/9391468577a1b96f180cd5c1f92341eaccf9515e). If you want to enable it for all the operating systems, please let me know.

Reply:
I was thinking for everything, but mac-only is fine for this PR since it focused on mac.